app/services/web_scraper.py

```diff
@@ -14,6 +28,54 @@ def scrape_webpage(url, timeout=15):
         }
         with requests.Session() as session:
             session.max_redirects = 5
+
+            # Special-case: Twitter / X tweets often require JS to render and
+            # return a placeholder page. Use Twitter's oEmbed endpoint as a
+            # reliable fallback to extract the tweet text and author.
+            parsed = urlparse(url)
+            hostname = parsed.hostname or ""
+            path = parsed.path or ""
+
+            if hostname.endswith("twitter.com") or hostname.endswith("x.com"):
+                # detect status (tweet) URLs
+                if "/status/" in path or "/statuses/" in path:
+                    try:
```

Comment:
Great work

1. Kindly remove all test files in this pr
2. The new functions don't have type hints while some existing code does. update all functions to be typed
3. _PLACEHOLDER_PATTERNS is duplicated in both web_scraper.py and ai_service.py. Should be in one place